init.lua

```diff
@@ -1,140 +1,153 @@
-                entity_name = string.sub(entity_name, index + 1)
-
-                death_message = string.gsub(death_message, '2', entity_name)
+        for _, player in ipairs(minetest.get_connected_players()) do
+            local ricevee_name = player:get_player_name()
+            local info = minetest.get_player_information(ricevee_name)
+            local lang = (info and info.lang_code) or "en"
+            local death_message = minetest.get_translated_string(lang, message)
+
+            death_message = string.gsub(death_message, '1', victim:get_player_name())
+            if killer then
+                if killer:is_player() then
+                    death_message = string.gsub(death_message, '2', killer:get_player_name())
+                else
+                    -- Get entity name, excluding mod name ("mymod:enemy" -> "enemy")
+                    entity_name = killer:get_luaentity().name
+                    index, _ = string.find(entity_name, ':')
+                    entity_name = string.sub(entity_name, index + 1)
+
+                    death_message = string.gsub(death_message, '2', entity_name)
+                end
             end
-        end
 
-        minetest.chat_send_all(death_message)
+            if ricevee_name then
+                minetest.chat_send_player(ricevee_name, death_message)
+            end
+        end
     end
 end
 
 minetest.register_on_dieplayer(function(player, reason)
+    -- Default cause: Not detected, send general death message
+    local message = get_death_message(other)
+
     if reason.object then
         if reason.object:is_player() then
             -- Player was killed by player
-            send_death_message(pvp, player, reason.object)
-        
+            message = get_death_message(pvp)
+
         else
             -- Player was killed by mob
-            send_death_message(mob, player, reason.object)
+            message = get_death_message(mob)
 
         end
     else
         if reason.type == 'fall' then
             -- Player was killed by fall damage
-            send_death_message(fall, player)
+            message = get_death_message(fall)
 
         elseif reason.type == 'drown' then
             -- Player drowned
-            send_death_message(drown, player)
+            message = get_death_message(drown)
 
         elseif reason.type == 'node_damage' then
             if string.match(reason.node, 'lava') then
                 -- Player burned in lava
-                send_death_message(lava, player)
+                message = get_death_message(lava)
 
             elseif string.match(reason.node, 'fire') then
                 -- Player burned in fire
-                send_death_message(burn, player)
+                message = get_death_message(burn)
 
-            else
-                -- Reason not detected, send general death message
-                send_death_message(other, player)
             end
-        else
-            send_death_message(other, player)
```

Comment:
This may be out of the scope of this PR.

Reply:
Yea, I thought it was necessary for server-side translated messages. Taken out!